docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+
+```php
+use Doctrine\ORM\Query\Expr\Join;
+use Shopsys\FrameworkBundle\Component\EntityExtension\QueryBuilder;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductCalculatedPrice;
+
+class MyProductFilterRepository
+{
+    // ...
+
+    /**
+     * @param \Doctrine\ORM\QueryBuilder $builder
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $limit
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup $group
+     */
+    protected function filterByMinimalPrice(QueryBuilder $builder, Money $limit, PricingGroup $group)
+    {
+        $condition = 'pcp.product = p AND pcp.pricingGroup = :group AND pcp.priceWithVat >= :limit';
+
+        $productsQueryBuilder->join(ProductCalculatedPrice::class, 'pcp', Join::WITH, $condition)
+            ->setParameter('group', $group)
+            ->setParameter('limit', $limit->getAmount());
+    }
+}
+```
+
+## Unit and Functional Tests
+
+You should use `Money` for all monetary values even in tests.
+
+Create the instances via `Money::create(int|string)` in your data providers or directly in your test methods if you don't use providers.
+
+You can use a custom PHPUnit constraint `IsMoneyEqual` for an assertion that two monetary values are equal using the `assertThat($value, Constraint $constraint)` method.
+
+Example:
+
+```php
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Tests\FrameworkBundle\Test\IsMoneyEqual;
+use Tests\ShopBundle\Test\FunctionalTestCase;
+
+class MyTest extends FunctionalTestCase
+{
+    // ...
+
+    /**
+     * @return array
+     */
+    public function customerLoyaltyCreditAmountProvider(): array
+    {
+        return [
+            [self::CUSTOMER_ID_WITHOUT_LOYALTY_CREDIT, Money::zero()],
+            [self::CUSTOMER_ID_WITH_LOW_LOYALTY_CREDIT, Money::create('12.5')],
+            [self::CUSTOMER_ID_WITH_HIGH_LOYALTY_CREDIT, Money::create(1000)],
+        ];
+    }
+
+    /**
+     * @dataProvider customerLoyaltyCreditAmountProvider
+     * @param int $customerId
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $expectedCreditAmount
+     */
+    public function testCustomerLoyaltyCreditAmount(int $customerId, Money $expectedCreditAmount) {
+        $customer = $this->getCustomerFromDatabase($customerId);
+
+        $creditAmount = $this->customerLoyaltyFacade->calculateCreditAmount($customer);
+
+        $this->assertThat($creditAmount, new IsMoneyEqual($expectedCreditAmount));
+    }
+}
+```
+
+## Price class
+
+`\Shopsys\FrameworkBundle\Model\Pricing\Price` is also an immutable [value object](https://codete.com/blog/value-objects/) used in pricing.
+
+It represents a price with and without VAT and is used in many parts of Shopsys Framework.
+Price calculation classes (except `InputPriceCalculation`) output instances of `Price`.
```

Comment:
You can maybe remove the sentence... I don't see added value and it also contains an exception that forces me to think about calculators while I should focus on `Price`

Reply:
Well, it's probably the most important usage, I think the mention is deserved.

Reply:
Ok, then I would suggest to remove the exception. As I said, it forces me to think about calculators while I should focus on `Price`... the exception interupts flow

Reply:
That's a fair point, I've removed the specific exception in 96d46fb74